src/modules/fw_att_control/FixedwingAttitudeControl.cpp

```diff
@@ -298,7 +298,9 @@ void
 FixedwingAttitudeControl::vehicle_manual_poll()
 {
 
-	if (_vcontrol_mode.flag_control_manual_enabled && !_vehicle_status.is_rotary_wing) {
+	// If VTOL, generate rates sp with FW attutide controller if in transition or in fixed-wing flight
+	if (_vcontrol_mode.flag_control_manual_enabled && !(_vehicle_status.is_rotary_wing &&
+			(!_vehicle_status.in_transition_mode || _parameters.vtol_type == vtol_type::TAILSITTER))) {
```

Comment:
The condition is too complex. Please break it up into multiple well described conditions first, e.g.:
```
const bool is_foo = ..;
const bool is_bar = ...;
if (is_foo && !is_bar) {
    ...
```

If you choose good names for the `bool`s, you can probably even remove the command above because the code is self explanatory.